instrumentor/runtimemigration/runtimemigration.go

```diff
@@ -0,0 +1,314 @@
+package runtimemigration
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"github.com/odigos-io/odigos/common/envOverwrite"
+	k8scontainer "github.com/odigos-io/odigos/k8sutils/pkg/container"
+	"github.com/odigos-io/odigos/k8sutils/pkg/envoverwrite"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MigrationRunnable struct {
+	KubeClient client.Client
+	Logger     logr.Logger
+}
+
+// This code ensures that migrationRunnable is categorized as an `Other` Runnable.
+func (m *MigrationRunnable) NeedLeaderElection() bool {
+	return false
+}
+
+func (m *MigrationRunnable) Start(ctx context.Context) error {
+
+	var instrumentationConfigs v1alpha1.InstrumentationConfigList
+	err := m.KubeClient.List(ctx, &instrumentationConfigs, &client.ListOptions{})
+	if err != nil {
+		m.Logger.Error(err, "Failed to list InstrumentationConfigs")
+		return nil
+	}
+
+	workloadNamespaces := make(map[string]map[string]map[string]*v1alpha1.InstrumentationConfig)
+	// Example structure:
+	// {
+	//   "deployment": {
+	//       "default": {
+	//           "frontend": *<InstrumentationConfig>,
+	//       },
+	//   },
+
+	for _, item := range instrumentationConfigs.Items {
+
+		IcName := item.GetName()
+		parts := strings.Split(IcName, "-")
+		if len(parts) < 2 {
+			continue
+		}
+
+		IcNamespace := item.GetNamespace()
+		workloadType := parts[0] // deployment/statefulset/daemonset
+		workloadName := strings.Join(parts[1:], "-")
+		if _, exists := workloadNamespaces[workloadType]; !exists {
+			workloadNamespaces[workloadType] = make(map[string]map[string]*v1alpha1.InstrumentationConfig)
+		}
```

Comment:
nit: this check can be removed if we initialize the map with entries for `deployment` `statefulset` and `daemonset`